src/content/subgames.cpp

```diff
@@ -283,35 +298,36 @@ std::vector<WorldSpec> getAvailableWorlds()
 	return worlds;
 }
 
+// bit of a hack but it's the simplest way to do this
+#define SETUP_SETTING(name) \
+	conf.set(#name, conf.get(#name))
+
 bool loadGameConfAndInitWorld(const std::string &path, const SubgameSpec &gamespec)
 {
-	// Override defaults with those provided by the game.
-	// We clear and reload the defaults because the defaults
-	// might have been overridden by other subgame config
-	// files that were loaded before.
-	g_settings->clearDefaults();
-	set_default_settings(g_settings);
-	Settings game_defaults;
-	getGameMinetestConfig(gamespec.path, game_defaults);
-	override_default_settings(g_settings, &game_defaults);
-
 	infostream << "Initializing world at " << path << std::endl;
 
 	fs::CreateAllDirs(path);
 
+	Settings conf;
+	set_default_settings(&conf);
+	override_default_settings(&conf, g_settings);
```

Comment:
was this meant to change order?

Reply:
yes, it initializes the default settings and then overrides them with the values from `minetest.conf`